### PR TITLE
Correctly handle NULL values in compound ART keys

### PR DESCRIPTION
**Pull request by taniabogatsch**
The current implementation of the `CREATE INDEX` operator does not handle `NULL` values in non-first columns of compound ART keys correctly. With this fix, we add another `PhysicalFilter` operator to the pipeline, which filters on 'IS NOT NULL' for each key column.

I also removed the check for `IsPrimary()`, which can never be true in the `CREATE INDEX` statement.

### CODE REVIEW

**Comment by Mytherin** [comment]
@lnkuiper can this be merged?

File: src/common/types/data_chunk.cpp [hunk 1]
```diff
@@ -92,6 +92,9 @@ bool DataChunk::AllConstant() const {
 }
 
 void DataChunk::Reference(DataChunk &chunk) {
+	if (chunk.ColumnCount() > ColumnCount()) {
+		D_ASSERT(chunk.ColumnCount() <= ColumnCount());
```
Comment (lnkuiper) [inline src/common/types/data_chunk.cpp:96]: this seems like debugging code, can we remove this?

**Review by lnkuiper — CHANGES_REQUESTED** [review]
I have one comment, then I think it's good to go:

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**